spiffworkflow-frontend/src/routes/ProcessInstanceShow.tsx

```diff
@@ -211,29 +225,125 @@ export default function ProcessInstanceShow() {
     }
   };
 
-  const taskDataDisplayArea = () => {
-    const taskToUse: any = taskToDisplay;
-    if (taskToDisplay) {
-      let createScriptUnitTestElement = null;
-      if (taskToUse.type === 'Script Task') {
-        createScriptUnitTestElement = (
+  const canEditTaskData = (task: any) => {
+    return task.state === 'READY';
+  };
+
+  const cancelEditingTaskData = () => {
+    setEditingTaskData(false);
+    initializeTaskDataToDisplay(taskToDisplay);
+  };
+
+  const taskDataStringToObject = (dataString: string) => {
+    return JSON.parse(dataString);
+  };
+
+  const saveTaskDataResult = (_: any) => {
+    setEditingTaskData(false);
+    const dataObject = taskDataStringToObject(taskDataToDisplay);
+    const taskToDisplayCopy = { ...taskToDisplay, data: dataObject }; // spread operator
+    setTaskToDisplay(taskToDisplayCopy);
+  };
+
+  const saveTaskDataFailure = (result: any) => {
+    console.log(result.toString());
+    // TODO: Not sure what to do here
```

Comment:
you could potentially just display a message to the user on failure using setErrorMessage, which is used in various components.